RileyLinkKitTests/PumpOpsSynchronousTests.swift

```diff
@@ -0,0 +1,480 @@
+        
+        loadSUT()
+    }
+    
+    func loadSUT() {
+        pumpState = PumpState(pumpID: pumpID, pumpRegion: pumpRegion)
+        pumpState.pumpModel = pumpModel
+        pumpState.awakeUntil = Date(timeIntervalSinceNow: 100) // pump is awake
+        
+        sut = PumpOpsSynchronous(pumpState: pumpState, session: rileyLinkCmdSession)
+        sut.communication = pumpOpsCommunicationStub
+    }
+    
+    func testShouldContinueIfTimestampBeforeStartDateNotEncountered() {
+        let pumpEvents: [PumpEvent] = [createBatteryEvent()]
+        
+        let (_, hasMoreEvents) = sut.convertPumpEventToTimestampedEvents(pumpEvents: pumpEvents, startDate: Date.distantPast, checkDate: Date.distantPast, pumpModel: pumpModel)
+        
+        XCTAssertTrue(hasMoreEvents)
+    }
+    
+    func testShouldFinishIfTimestampBeforeStartDateEncountered() {
+        let batteryEvent = createBatteryEvent()
+        let pumpEvents: [PumpEvent] = [batteryEvent]
+        
+        let afterBatteryEventDate = batteryEvent.timestamp.date!.addingTimeInterval(TimeInterval(hours: 10))
+        
+        let (_, hasMoreEvents) = sut.convertPumpEventToTimestampedEvents(pumpEvents: pumpEvents, startDate: afterBatteryEventDate, checkDate: Date.distantPast, pumpModel: pumpModel)
+        
+        XCTAssertFalse(hasMoreEvents)
+    }
+    
+    func testBatteryEventIsCreated() {
+        let pumpEvents: [PumpEvent] = [createBatteryEvent()]
+        
+        let (events, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: pumpEvents, startDate: Date.distantPast, checkDate: Date.distantPast, pumpModel: pumpModel)
+        
+        XCTAssertEqual(events.count, 1)
+    }
+    
+    func testMultipleBatteryEvent() {
+        
+        let batteryEvent2007 = createBatteryEvent(withDateComponent: dateComponents2017)
+        let batteryEvent2017 = createBatteryEvent(withDateComponent: dateComponents2007)
+        let pumpEvents: [PumpEvent] = [batteryEvent2007, batteryEvent2017]
+        
+        let (events, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: pumpEvents, startDate: Date.distantPast, checkDate: Date(), pumpModel: pumpModel)
+        
+        XCTAssertEqual(events.count, 2)
+    }
+
+    func testOldBatteryEventIsFiltered() {
+        
+        let datePast2007 = dateComponents2007.date!.addingTimeInterval(60*60)
+        let datePassed2017 = dateComponents2017.date!.addingTimeInterval(60*60)
+        
+        let batteryEvent2007 = createBatteryEvent(withDateComponent: dateComponents2017)
+        let batteryEvent2017 = createBatteryEvent(withDateComponent: dateComponents2007)
+        let pumpEvents: [PumpEvent] = [batteryEvent2007, batteryEvent2017]
+        
+        let (events, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: pumpEvents, startDate: datePast2007, checkDate: datePassed2017, pumpModel: pumpModel)
+        
+        XCTAssertEqual(events.count, 1)
+    }
+    
+    func testOutOfOrderEventReturnsCancel() {
```

Comment:
Perhaps change the name of these tests to avoid using the phrase 'out of order event', as it's easy to conflate with the delayed append events, like square wave bolus on 522 (which are also out of order).  The code that these tests are testing is to check for a case of date corruption on the pump. So maybe something like testDateCorruptionReturnsCancel()